packages/react-components/react-card/src/components/Card/useCard.ts

```diff
@@ -28,13 +28,26 @@ export const useCard_unstable = (props: CardProps, ref: React.Ref<HTMLElement>):
 
   const focusAttrs = focusMode !== 'off' ? { tabIndex: 0, ...groupperAttrs } : null;
 
+  const isInteractive = Boolean(
+    ['a', 'button'].includes(as) ||
+      props.onClick ||
+      props.onDoubleClick ||
+      props.onMouseUp ||
+      props.onMouseDown ||
+      props.onPointerUp ||
+      props.onPointerDown ||
+      props.onTouchStart ||
+      props.onTouchEnd,
+  );
+
   return {
     appearance,
     orientation,
     size,
+    isInteractive,
 
-    components: { root: 'div' },
-    root: getNativeElementProps(props.as || 'div', {
+    components: { root: as },
+    root: getNativeElementProps(as, {
```

Comment:
in the case of `'a'` we should perhaps consider the possibility of adding `role=button` and treat `'a'` with the same navigation methods of a button (`Enter` and `Space` key events should invoke clicks), to ensure accessibility. Have a look at `useARIAButton` hooks to add that behaviour conditionally if desired.

Reply:
The guidance for card is to have `role=group`. If we have some cards side by side and only one is a `button` and we change the `role` for that card, it'll be wrong. That's why we keep the `role=group` in this case.